zipkin-server/src/main/java/zipkin/server/ActuatorConditionalImporter.java

```diff
@@ -0,0 +1,73 @@
+/*
+ * Copyright 2015-2019 The OpenZipkin Authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License"); you may not use this file except
+ * in compliance with the License. You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software distributed under the License
+ * is distributed on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express
+ * or implied. See the License for the specific language governing permissions and limitations under
+ * the License.
+ */
+package zipkin.server;
+
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.boot.autoconfigure.EnableAutoConfiguration;
+import org.springframework.boot.context.properties.bind.Binder;
+import org.springframework.context.ApplicationContextInitializer;
+import org.springframework.context.support.GenericApplicationContext;
+import org.springframework.core.env.ConfigurableEnvironment;
+
+/**
+ * Actuator relies on classpath scanning, so we cannot choose one configuration entrypoint unless
+ * using {@link EnableAutoConfiguration}. We disable scanning in {@link ZipkinServer} by default so
+ * this conditionally loads actuator a different way.
+ *
+ * <p>This queries the property {@link #PROPERTY_NAME_ACTUATOR_INCLUDE} which cherry-picks the most
+ * relevant configuration for when actuator is in the classpath.
+ */
+final class ActuatorConditionalImporter
+  implements ApplicationContextInitializer<GenericApplicationContext> {
```

Comment:
This feels like an overly complex solution when we are already using `@Configuration` and `@Conditional` is available. Couldn't you just write a `@Configuration` that did this?

Reply:
The main problem in actuator is the types are needed. we have another pull request that makes it possible to remove actuator (at compile time meaning actuator is not present)

Like mentioned before there's no way I know of now anyway to get actuator to load when you turn off autoconfiguration.

Reply:
I expect you would have to explicitly list all the configurations you need. But that's a sacrifice you already made, and you could do it in a class that was marked as `@Conditional` right?

Reply:
If I had a type that was conditional, I'd still need a way to load the types mentioned. For example, bean methods are not used here, we are doing things like imports, and imports requires strict types, not class names. ex  I could if there was somethign like `@Import(name="fully.qualified.type")` but I scrubbed and couldn't find one. Looking at your bench repo literally was how I discovered this other way. I'm glad I found that as before it was a hard stop.

Reply:
I think what you are looking for is a `ImportSelector` or `ImportBeanDefinitionRegistrar` (both mentioned in the javadocs for `@Import`).

Reply:
I'll make a note that someone can figure out how to use those things and refactor